android/RNKeys.gradle

```diff
@@ -5,11 +5,20 @@ import java.util.regex.Pattern
 
 def getCurrentFlavor() {
   Gradle gradle = getGradle()
+  String taskName = gradle.getStartParameter().getTaskNames()[0]
 
   // match optional modules followed by the task
   // (?:.*:)* is a non-capturing group to skip any :foo:bar: if they exist
   // *[a-z]+([A-Za-z]+) will capture the flavor part of the task name onward (e.g., assembleRelease --> Release)
   def pattern = Pattern.compile("(?:.*:)*[a-z]+([A-Z][A-Za-z0-9]+)")
+
+  if (taskName?.contains("assemble"))
```

Comment:
Good catch 🙌 we should remove duplication, you can remove duplication like this
```
if (taskName?.startsWith("assemble") || taskName?.startsWith("bundle")) {
    //e.g. if assemblestagingrelease or bundlestagingrelease then flavor staging
    pattern = Pattern.compile("(?:assemble|bundle)(\\w+)([Rr]elease|[Dd]ebug)")
  }
```

Reply:
sure 👍🏻